l10n_ch_invoice_reports/models/mail_template.py

```diff
@@ -14,49 +14,33 @@ def generate_email(self, res_ids, fields=None):
 
         result = super().generate_email(res_ids, fields)
-        if self._name == "account.move":
-            multi_mode = True
-            if isinstance(res_ids, int):
-                res_ids = [res_ids]
-                multi_mode = False
-
+        if self._should_print_invoice_attachment():
             res_ids_to_templates = self.get_email_template(res_ids)
-            for res_id in res_ids:
-                related_record = self.env[self.model_id.model].browse(res_id)
-                template = res_ids_to_templates[res_id]
-                self.generate_invoice_attachment(
-                    result, template, res_id, related_record, multi_mode
+            for res_id in self.env["account.move"].browse(res_ids):
+                related_record = self.env[self.model_id.model].browse(res_id.id)
+                template = res_ids_to_templates[res_id.id]
+                invoice_attachments = self.generate_invoice_attachment(
+                    template, res_id.id, related_record
                 )
+                result[res_id.id]["attachments"] = invoice_attachments
         return result
 
-    def generate_invoice_attachment(
-        self, result, template, res_id, related_record, multi_mode
-    ):
-        result[res_id]["attachments"] = False
+    def _should_print_invoice_attachment(self):
+        if (
+            self.model_id.model == "account.move"
+            and self.report_template.report_name
+            == "l10n_ch_invoice_reports.account_move_payment_report"
+        ):
+            return True
+
+    def generate_invoice_attachment(self, template, res_id, related_record):
         self._render_template(template.report_name, template.model, res_id)
-        new_attachments = []
-        # We add an optional attachment from mail template (if set)
-        if self.report_template:
-            report_name = self.report_template.attachment
-            if (
-                self.report_template.report_name
-                == "l10n_ch_invoice_reports.account_move_payment_report"
-            ):
-                # force translation of attachment's name according to partner's lang
-                template = self.with_context(lang=related_record.partner_id.lang)
-                report_name = _(
-                    "invoice_%s_with_payslip.pdf"
-                ) % related_record.name.replace("/", "_")
-            report_xml_id = self.report_template.xml_id
-            report_pdf = self.env.ref(report_xml_id).render_qweb_pdf([res_id])[0]
-            report_pdf = base64.b64encode(report_pdf)
-            new_attachments.append((report_name, report_pdf))
-        attachments_list = (
-            result[res_id].get("attachments", [])
-            if multi_mode
-            else result.get("attachments", [])
+        # force translation of attachment's name according to partner's lang
+        self = self.with_context(lang=related_record.partner_id.lang)
```

Comment:
@simahawk  tried to keep this template_obj = self.with_context(lang=related_record.partner_id.lang)
and then use template_obj instead of self in lines below but doesn't work (or i misunderstood something here). Anyway - pdb shows lang in context  = en_US which is probably admin user lang, not the invoice partner lang. if you have some particular construction in your head, just comment here. I don't want to do more trial and errors with this.